graph/manifest.go

```diff
@@ -75,12 +75,15 @@ func (s *TagStore) newManifest(localName, remoteName, tag string) ([]byte, error
 
 			tarSum, err := tarsum.NewTarSum(archive, true, tarsum.Version1)
 			if err != nil {
+				archive.Close()
```

Comment:
I had thought about using "defer" but I found that it is little tricky here. First of all I can't use a general defer to close it as we are using a for loop inside the function which runs multiple times and archive. Close() needs to happen for each layer.

Now I should be able to use defer in such a way so that it closes archive only in case of error then it can happen that we call close() on archive twice. So say first we do archive.Close() in success path and later failure happens due to some other reason in same function. In that case we will end up calling archive.Close() again. I am not sure if calling Close() on already closed archive is a problem or not.

Is there a way to undefer already scheduled defer?